### PR TITLE
feat(all): moved EnvService to commons + exposed getXrayTraceId in tracer

### LINKED ISSUE #1070: Feature request: access to the root trace id in Tracer
## Description of the feature request

**Problem statement**
When handling application errors, I want to embed the XRay Trace ID in the payload & error message returned by my API so that customers can cite the trace ID in support requests, and so our development team can easily look up the trace from a customer service request via AWS XRay's web console.

**Summary of the feature**
I'd like the tracer to provide the root trace id so that I can access it in my (otherwise) unhandled exceptions handler when I construct a user-facing error message.

**Code examples**

```ts
const rootTraceId = tracer.getRootTraceId();

// Example of returning an error with Express.js
res.status(500).json({
  error: `Internal Error - Please contact support and quote the following id: ${rootTraceId}`,
  // Include the rootTraceId in the response so we can show a "contact support" button that
  // takes the customer to a customer service form with the trace as additional context.
  rootTraceId, 
});
```

**Benefits for you and the wider AWS community**
Makes it easier for the wider community to expose trace IDs to their customers so that developers can look up the conditions and contexts of the errors reported by their customers.

**Describe alternatives you've considered**
For now, we've implemented this by parsing out the `Root=` traceid from `process.env['_X_AMZN_TRACE_ID']`, but our preference would be to depend on Powertools for this.

```ts
export function getTraceId(): string | undefined {
  const traceId = process.env['_X_AMZN_TRACE_ID'];
  if (!traceId) return;

  return parseTraceId(traceId);
}

export function parseTraceId(traceId: string): string | undefined {
  if (traceId.includes('Root=')) {
    // Example:
    // Root=1-6303d71b-63c75c564e336c9c2cffc1d5;Parent=58e97d4cc1ed1447;Sampled=1
    const [rootTraceId] = traceId
      .split(';')
      .filter((segment) => segment.startsWith('Root='))
      .map((segment) => {
        const [_, rootTraceId] = segment.split('=');
        return rootTraceId;
      });

    if (rootTraceId) {
      // 1-6303d71b-63c75c564e336c9c2cffc1d5
      return rootTraceId;
    }
  }

  return traceId;
}
```

**Additional context**
None

### Related issues, RFCs
None

### PR DESCRIPTION
**Pull request by dreamorosi**
## Description of your changes

In #1070 customers have requested the addition of a new method in Tracer that allows to access the AWS X-Ray Root Trace ID. This is useful in those situations in which customers want to use this id as correlation id in the responses returned by a function or manually pass it to downstream processes.

The logic to retrieve and parse this ID was already present in Logger ([here](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/packages/logger/src/Logger.ts#L559-L563)). For that utility we use the method to append the `xray_trace_id` to log entries [in some cases](https://awslabs.github.io/aws-lambda-powertools-typescript/latest/core/logger/#standard-structured-keys).

During the discussion of the feature it became clear that this could be a good moment to also extract this logic to the `commons` package continuing the work started in #484, and potentially unlocking/facilitating #165.

#### Before this change
image

#### After this change
image

This issue also introduces a new public method in Tracer called `getRootXrayTraceId` that allows customers to access the AWS X-Ray Root Trace ID within a Lambda handler.

Fixes #1070 

### How to verify this change

Check the unit tests results, as well as package size report, and e2e tests.

### Related issues, RFCs

**Issue number:** #1070

### PR status

***Is this ready for review?:*** YES  
***Is it a breaking change?:*** NO

## Checklist

- [x] [My changes meet the tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
- [x] I have performed a *self-review* of my own code
- [x] I have *commented* my code where necessary, particularly in areas that should be flagged with a TODO, or hard-to-understand areas
- [x] I have made corresponding changes to the *documentation*
- [ ] I have made corresponding changes to the *examples*
- [x] My changes generate *no new warnings*
- [x] The *code coverage* hasn't decreased
- [x] I have *added tests* that prove my change is effective and works
- [x] New and existing *unit tests pass* locally and in Github Actions
- [ ] Any *dependent changes have been merged and published* in downstream module
- [x] The PR title follows the [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-typescript/blob/main/.github/semantic.yml#L2)

### Breaking change checklist

N/A

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by dreamorosi** [comment]
E2E tests run: https://github.com/awslabs/aws-lambda-powertools-typescript/actions/runs/3271574957

File: docs/core/tracer.md [hunk 1]
```diff
@@ -484,6 +484,38 @@ Use **`POWERTOOLS_TRACER_CAPTURE_ERROR=false`** environment variable to instruct
 
     1. You might **return sensitive** information from exceptions, stack traces you might not control
 
+### Access AWS X-Ray Root Trace ID
+
+Tracer exposes a `getRootXrayTraceId()` method that allows you to retrieve the [AWS X-Ray Root Trace ID](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-traces) corresponds to the current function execution.
+
+!!! info "This is commonly useful in two scenarios"
+
+    1. By including the root trace id in your response consumers can use it to correlate requests
```
Comment (misterjoshua) [inline docs/core/tracer.md:493]:
```suggestion
    1. By including the root trace id in your response consumers, can use it to correlate requests
```

File: packages/logger/src/config/EnvironmentVariablesService.ts [hunk 2]
```diff
@@ -123,7 +120,7 @@ class EnvironmentVariablesService extends ConfigService {
    * @returns {string}
    */
   public getXrayTraceId(): string {
-    return this.get(this.xRayTraceIdVariable);
+    return super.getXrayTraceId();
```
Comment (misterjoshua) [inline packages/logger/src/config/EnvironmentVariablesService.ts:123]: Given that this class extends commons' `EnvironmentVariablesService` and that the commons superclass marks `getXrayTraceId()` as public, do we need to define a `getXrayTraceId()` method here?

File: packages/tracer/src/config/EnvironmentVariablesService.ts [hunk 3]
```diff
@@ -40,7 +44,7 @@ class EnvironmentVariablesService extends ConfigService {
   }
 
   public getXrayTraceId(): string {
-    return this.get(this.xRayTraceIdVariable);
+    return super.getXrayTraceId();
```
Comment (misterjoshua) [inline packages/tracer/src/config/EnvironmentVariablesService.ts:47]: Same as for `EnvironmentVariablesService` in logger, are we sure we need this?

**Review by misterjoshua — COMMENTED** [review]
This PR looks pretty good to me. For now, I have just a few minor comments.

File: packages/logger/src/config/EnvironmentVariablesService.ts (hunk 2 above)
Comment (dreamorosi) [reply packages/logger/src/config/EnvironmentVariablesService.ts]: Very good spot @misterjoshua, I missed this. 

I have also removed the extra `getServiceName` one from all.

File: packages/tracer/src/config/EnvironmentVariablesService.ts (hunk 3 above)
Comment (dreamorosi) [reply packages/tracer/src/config/EnvironmentVariablesService.ts]: Thanks, removed, see here https://github.com/awslabs/aws-lambda-powertools-typescript/pull/1123#discussion_r999231243

**Review by misterjoshua — DISMISSED** [review]
This all looks good to me!

**Review by saragerion — DISMISSED** [review]
LGTM!

File: packages/commons/tests/unit/config/EnvironmentVariablesService.test.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+      // Assess
+      expect(value).toEqual('shopping-cart-api');
+    });
+
+  });
+
+  describe('Method: getXrayTraceId', () => {
+
+    test('It returns the value of the environment variable _X_AMZN_TRACE_ID', () => {
```
Comment (flochaz) [inline packages/commons/tests/unit/config/EnvironmentVariablesService.test.ts:73]: can we add better test for this logic: 
1. if empty ?
2. if real (see the split works)

**Review by flochaz — APPROVED** [review]
LGTM, minor update on UT would b great

**Review by flochaz — APPROVED** [review]

**Pull request merged**